### PR TITLE
Replace deprecated Assert.assertThat(...) with MatcherAssert.assertTh…

### PR DESCRIPTION
**Pull request by normanmaurer**
…at(...)

Motivation:

junit deprecated Assert.assertThat(...)

Modifications:

Use MatcherAssert.assertThat(...) as replacement for deprecated method

Result:

Less deprecation warnings

### CODE REVIEW

**Comment by normanmaurer** [comment]
See https://github.com/junit-team/junit4/blob/HEAD/doc/ReleaseNotes4.13.md#pull-request-1150-deprecate-assertassertthat

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: buffer/src/test/java/io/netty/buffer/ByteBufDerivationTest.java [hunk 1]
```diff
@@ -21,8 +21,8 @@
 import java.nio.ByteOrder;
 import java.util.Random;
 
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.*;
-import static org.junit.Assert.*;
```
Comment (chrisvest) [inline buffer/src/test/java/io/netty/buffer/ByteBufDerivationTest.java:25]: What's our preference on star-imports? I thought we were avoiding them everywhere. If so, we could "organise import" on the files already touched by this PR.

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketCloseStatusTest.java [hunk 2]
```diff
@@ -64,12 +65,12 @@ public void testKnownStatuses() {
...
-        Assert.assertThat(PROTOCOL_ERROR, Matchers.comparesEqualTo(PROTOCOL_ERROR));
-        Assert.assertThat(PROTOCOL_ERROR, Matchers.comparesEqualTo(valueOf(1002)));
-        Assert.assertThat(PROTOCOL_ERROR, Matchers.lessThan(INVALID_MESSAGE_TYPE));
-        Assert.assertThat(PROTOCOL_ERROR, Matchers.lessThan(valueOf(1007)));
+        MatcherAssert.assertThat(PROTOCOL_ERROR, Matchers.greaterThan(NORMAL_CLOSURE));
+        MatcherAssert.assertThat(PROTOCOL_ERROR, Matchers.greaterThan(valueOf(1001)));
+        MatcherAssert.assertThat(PROTOCOL_ERROR, Matchers.comparesEqualTo(PROTOCOL_ERROR));
+        MatcherAssert.assertThat(PROTOCOL_ERROR, Matchers.comparesEqualTo(valueOf(1002)));
+        MatcherAssert.assertThat(PROTOCOL_ERROR, Matchers.lessThan(INVALID_MESSAGE_TYPE));
+        MatcherAssert.assertThat(PROTOCOL_ERROR, Matchers.lessThan(valueOf(1007)));
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketCloseStatusTest.java:73]: Static import?

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketServerHandshaker13Test.java [hunk 3]
```diff
@@ -103,7 +104,7 @@ private static void testCloseReason0(ChannelHandler... handlers) {
             // expected
         }
         ReferenceCounted closeMessage = ch.readOutbound();
-        Assert.assertThat(closeMessage, CoreMatchers.instanceOf(ByteBuf.class));
+        MatcherAssert.assertThat(closeMessage, CoreMatchers.instanceOf(ByteBuf.class));
```
Comment (chrisvest) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketServerHandshaker13Test.java:107]: Static import?

File: resolver-dns-native-macos/src/test/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProviderTest.java [hunk 4]
```diff
@@ -45,7 +46,7 @@ public void testStream() {
 
     @Test
     public void testDefaultUseCorrectInstance() {
-        Assert.assertThat(DnsServerAddressStreamProviders.platformDefault(),
+        MatcherAssert.assertThat(DnsServerAddressStreamProviders.platformDefault(),
```
Comment (chrisvest) [inline resolver-dns-native-macos/src/test/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProviderTest.java:49]: Static import?

File: resolver-dns/src/test/java/io/netty/resolver/dns/DnsAddressResolverGroupTest.java [hunk 5]
```diff
@@ -48,7 +49,8 @@ public void testUseConfiguredEventLoop() throws InterruptedException {
                 @Override
                 public void operationComplete(Future<Object> future) {
                     try {
-                        Assert.assertThat(future.cause(), Matchers.instanceOf(UnsupportedAddressTypeException.class));
+                        MatcherAssert.assertThat(future.cause(),
+                                Matchers.instanceOf(UnsupportedAddressTypeException.class));
```
Comment (chrisvest) [inline resolver-dns/src/test/java/io/netty/resolver/dns/DnsAddressResolverGroupTest.java:53]: Static imports?

File: transport/src/test/java/io/netty/channel/nio/NioEventLoopTest.java [hunk 6]
```diff
@@ -222,7 +223,7 @@ public void run() {
             group.shutdownNow();
             t.join();
             group.terminationFuture().syncUninterruptibly();
-            assertThat(error.get(), IsInstanceOf.instanceOf(RejectedExecutionException.class));
+            assertThat(error.get(), Matchers.instanceOf(RejectedExecutionException.class));
```
Comment (chrisvest) [inline transport/src/test/java/io/netty/channel/nio/NioEventLoopTest.java:226]: Static import?

File: buffer/src/test/java/io/netty/buffer/ByteBufDerivationTest.java (hunk 1 above)
Comment (normanmaurer) [reply buffer/src/test/java/io/netty/buffer/ByteBufDerivationTest.java:25]: I actually would like to remove these but I would do this in a followup pr

Comment (chrisvest) [reply buffer/src/test/java/io/netty/buffer/ByteBufDerivationTest.java:25]: Works for me 👍

**Review by chrisvest — APPROVED** [review]

**Pull request merged**